### PR TITLE
Merged Yusesope's code into dev-v2 base

### PR DESCRIPTION
**Pull request by steffenmanden**
This should merge it properly to the dev-2 base

### CODE REVIEW

File: library/extractor/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java [hunk 1]
```diff
@@ -1269,6 +1269,9 @@ protected void binaryElement(int id, int contentSize, ExtractorInput input) thro
         handleBlockAdditionalData(
             tracks.get(blockTrackNumber), blockAdditionalId, input, contentSize);
         break;
+      case ID_BLOCK_ADD_ID_EXTRA_DATA:
```
Comment (yusesope) [inline library/extractor/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java:1270]: @steffenmanden I think you missed this part in my original code ;)
Check for:
```
case ID_BLOCK_ADD_ID_EXTRA_DATA:
	currentAdditionMapping.extraData = new byte[contentSize];
	input.readFully(currentAdditionMapping.extraData, 0, contentSize);
	Log.i("yusesope",String.format("ID_BLOCK_ADD_ID_EXTRA_DATA size = %d", currentAdditionMapping.extraData.length));
```
See there was one of my log lines you hate so much!
I need them to debug the code ;)
Then I delete them in one go with a regExp:
`Log.\w\("yusesope".*?;\n`

File: library/extractor/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java [hunk 2]
```diff
@@ -1270,7 +1270,7 @@ protected void binaryElement(int id, int contentSize, ExtractorInput input) thro
             tracks.get(blockTrackNumber), blockAdditionalId, input, contentSize);
         break;
       case ID_BLOCK_ADD_ID_EXTRA_DATA:
-        currentAdditionMapping.extraData = new byte[contentSize]
+        currentAdditionMapping.extraData = new byte[contentSize];
```
Comment (yusesope) [inline library/extractor/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java]: 👍

File: library/extractor/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java [hunk 3]
```diff
@@ -1967,12 +2010,44 @@ public void initializeOutput(ExtractorOutput output, int trackId) throws ParserE
...
           nalUnitLengthFieldLength = hevcConfig.nalUnitLengthFieldLength;
+          if (listAdditionMapping != null){
+            for (AdditionMapping addMappHevc : listAdditionMapping) {
+              switch (addMappHevc.idType) {
+                case AdditionMapping.TYPE_dvcC:
+                  DolbyVisionConfig dolbyVisionConfig = DolbyVisionConfig.parse(
+                      new ParsableByteArray(addMappHevc.extraData)
+                  );
+                  if (dolbyVisionConfig != null) {
+                    codecs = dolbyVisionConfig.codecs;
```
Comment (yusesope) [inline library/extractor/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java:2039]: @steffenmanden you forgot to initialize `codecs`:
`String codecs = null;`

EDIT:
My bad, I found it :).
Please also ignore my comment

File: library/extractor/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java (hunk 1 above)
Comment (steffenmanden) [reply library/extractor/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java:1270]: I know we use Logs - usually i keep them locally and dont commit them - if you prefer to have them there its OK :D as long as you promise we will clean up

Comment (yusesope) [reply library/extractor/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java:1270]: No problem. I will do the same!
It was just to explain why :)!

Comment (steffenmanden) [reply library/extractor/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java:1270]: But now you can see why Github is nice - we can see each others code changes and comment them - and then make a shared codebase :)

**Comment by steffenmanden** [comment]
If you agree with the changes, you should be able to merge them in :)

**Pull request merged**